core/trino-spi/src/main/java/io/trino/spi/block/RowBlockBuilder.java

```diff
@@ -64,6 +65,7 @@ private RowBlockBuilder(@Nullable BlockBuilderStatus blockBuilderStatus, BlockBu
         this.fieldBlockOffsets = requireNonNull(fieldBlockOffsets, "fieldBlockOffsets is null");
         this.rowIsNull = requireNonNull(rowIsNull, "rowIsNull is null");
         this.fieldBlockBuilders = requireNonNull(fieldBlockBuilders, "fieldBlockBuilders is null");
+        this.singleRowBlockWriter = new SingleRowBlockWriter(fieldBlockBuilders[0].getPositionCount(), fieldBlockBuilders);
```

Comment:
Initial position should be invalid one (e.g -1) so that one cannot use `singleRowBlockWriter` without calling `beginBlockEntry` first

Reply:
In current implementation it is not possible to use `singleBlockWriter` without calling `beginBlockEntry` (It is encapsulated). For my eye this preventions is sufficient. But, if you think that it is necessary, I can initiailze it with `-1`

Reply:
> In current implementation it is not possible to use singleBlockWriter without calling beginBlockEntry (It is encapsulated). For my eye this kind of prevention is sufficient. But, if you think that it is necessary, I can initiailze it with -1

One could keep `singleBlockWriter` after close entry and use it

Reply:
ok